lib/MetaCPAN/Web/Model/ReleaseInfo.pm

```diff
@@ -165,6 +167,51 @@ sub normalize {
     };
 }
 
+sub groom_repository {
+    my ( $self, $release ) = @_;
+    my $repo = $release->{resources}{repository}
+        or return {};
+
+    $repo = {%$repo};
+
+    for my $type (qw(url web)) {
+        my $url = $repo->{$type}
+            or next;
+
+        if (
+            $url =~ m{
+            \A
+            (?:ssh|https?|git)://
```

Comment:
Had you thought about using `URI` here? It supports `git`. https://metacpan.org/pod/URI::git

Reply:
I could, but I'm not sure it would really make things any more clear. The regex would end up split into two, and various extra checks would have to be added.